src/Chapter2.hs

```diff
@@ -858,7 +892,8 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+rewind :: [a] -> [a]
+rewind = foldl (flip (:)) []
```

Comment:
Clever solution!
You could also implemented it with the pattern matching, but what you did is great 💯